File: samples/other-languages/crash.c
/*
Sums grades into an array.  Bug: the array pointer is never given any memory,
so the first write crashes.  autotester should say: crash (SIGSEGV).

Example execution 1:
How many grades? -> 2
Grade: -> 80
Grade: -> 90
Total: 170
*/
#include <stdio.h>
#include <stdlib.h>
int main(void) {
    int n, i, total = 0;
    int *grades = NULL;          /* should have been malloc'd */
    printf("How many grades? ");
    scanf("%d", &n);
    for (i = 0; i < n; i++) {
        printf("Grade: ");
        scanf("%d", &grades[i]);
        total += grades[i];
    }
    printf("Total: %d\n", total);
    return 0;
}

File: samples/python/crash.py
"""
Average of a list of grades.  Bug: dividing by zero when no grades are entered.
autotester should say: crash, ZeroDivisionError.

Example execution 1:
Grade (or -1 to stop): -> 80
Grade (or -1 to stop): -> 90
Grade (or -1 to stop): -> -1
Average: 85.0

Example execution 2:
Grade (or -1 to stop): -> -1
Average: 0.0
"""
total = count = 0
while True:
    g = int(input("Grade (or -1 to stop): "))
    if g == -1:
        break
    total += g
    count += 1
print(f"Average: {total / count}")
